apps/web/src/pages/templates/workflow/DigestMetadata.tsx

```diff
@@ -1,213 +1,427 @@
-import { Input, SegmentedControl, Switch } from '../../../design-system';
+import { colors, Input, SegmentedControl, Select, Tooltip } from '../../../design-system';
 import { inputStyles } from '../../../design-system/config/inputs.styles';
-import { useEnvController } from '../../../hooks';
-import { IntervalRadios } from './IntervalRadios';
-import { LabelWithTooltip } from './LabelWithTooltip';
+import { WeekDaySelect } from './digest/WeekDaySelect';
+import { Bell, Digest, Timer } from '../../../design-system/icons';
+import { TypeSegmented } from './digest/TypeSegment';
+import { IntervalSelect } from './digest/IntervalSelect';
+import { BackOffFields } from './digest/BackOffFields';
+import { WillBeSentHeader } from './digest/WillBeSentHeader';
+import { ScheduleMonthlyFields } from './digest/ScheduleMonthlyFields';
+import { useEffect } from 'react';
+import { RegularInfo } from './digest/icons/RegularInfo';
+import { format } from 'date-fns';
 
-const StyledSwitch = styled(Switch)`
-  max-width: 100% !important;
-  margin-top: 15px;
-`;
+const convertUnitToLabel = (unit: DigestUnitEnum) => {
+  switch (unit) {
+    case DigestUnitEnum.SECONDS:
+      return 'second';
+    case DigestUnitEnum.MINUTES:
+      return 'minute';
+    case DigestUnitEnum.HOURS:
+      return 'hour';
+    case DigestUnitEnum.DAYS:
+      return 'day';
+    case DigestUnitEnum.WEEKS:
+      return 'week';
+    case DigestUnitEnum.MONTHS:
+      return 'month';
+  }
+};
 
-export const DigestMetadata = ({ control, index }) => {
-  const { readonly } = useEnvController();
+export const DigestMetadata = ({ control, index, readonly }) => {
   const {
-    formState: { errors, isSubmitted, isDirty },
+    formState: { errors, isSubmitted },
     watch,
     trigger,
+    setValue,
   } = useFormContext();
 
+  const { colorScheme } = useMantineColorScheme();
+
   const type = watch(`steps.${index}.metadata.type`);
+  const unit = watch(`steps.${index}.metadata.unit`);
+  const amount = watch(`steps.${index}.metadata.amount`);
+  const digestKey = watch(`steps.${index}.metadata.digestKey`);
   const showErrors = isSubmitted && errors?.steps;
+  const isEnterprise = process.env.REACT_APP_DOCKER_HOSTED_ENV !== 'true';
+
+  useEffect(() => {
+    if (type === DigestTypeEnum.TIMED && unit === DigestUnitEnum.MINUTES) {
+      setValue(`steps.${index}.metadata.unit`, DigestUnitEnum.DAYS);
+    }
+
+    if (type === DigestTypeEnum.REGULAR && unit === DigestUnitEnum.DAYS) {
+      setValue(`steps.${index}.metadata.unit`, DigestUnitEnum.MINUTES);
+    }
+
+    trigger(`steps.${index}.metadata`);
+  }, [unit, type]);
+
+  useEffect(() => {
+    if (type === DigestTypeEnum.TIMED && amount === 5) {
+      setValue(`steps.${index}.metadata.amount`, 1);
+    }
+
+    if (type === DigestTypeEnum.REGULAR && amount === 1) {
+      setValue(`steps.${index}.metadata.amount`, 5);
+    }
+
+    trigger(`steps.${index}.metadata`);
+  }, [type, amount]);
```

Comment:
I am struggling to understand what's going on in these 2 effects. Mostly because it doesn't feel natural to me we are swapping those values.
What's the goal of these 2 effects?

Reply:
To change the defaults when user changes the type of digest